eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ErrorHandlers.scala

```diff
@@ -287,7 +288,7 @@ trait ErrorHandlers extends CommonHandlers {
 
   def handleRemoteSpentOther(tx: Transaction, d: PersistentChannelData) = {
     log.warning(s"funding tx spent in txid=${tx.txid}")
-    Closing.RevokedClose.claimCommitTxOutputs(keyManager, d.commitments, tx, nodeParams.db.channels, nodeParams.onChainFeeConf.feeEstimator, nodeParams.onChainFeeConf.feeTargets) match {
+    Try(Closing.RevokedClose.claimCommitTxOutputs(keyManager, d.commitments, tx, nodeParams.db.channels, nodeParams.onChainFeeConf.feeEstimator, nodeParams.onChainFeeConf.feeTargets)).toOption.flatten match {
```

Comment:
This is a shortcut, I'd rather slightly rework `Closing.RevokedClose.claimCommitTxOutputs` to make sure it never throws: I believe we only need to remove the two `require` and replace them by returning `None` if their conditions aren't verified. That would be cleaner, wouldn't it?

Reply:
> That would be cleaner, wouldn't it?

Taken in isolation, yes, but then it would be inconsistent with other `*Closed.claim*` methods which all assume that they are called for the correct tx and may throw.

I think it could make sense to have all those methods return a `Try`, but then it's a bigger change.

Reply:
What about first checking whether the tx is a revoked commit before calling `Closing.RevokedClose.claimCommitTxOutputs`? That would be easier to read and it would be more obvious that the two "other" cases are either a revoked commitment or something completely unknown.

Reply:
I just realize now that `RevokedClose.claimCommitTxOutputs` returns an `Option`, so my earlier point about creating an inconsistency is moot . I reworked it as per your initial suggestion in 238f51f67.

Reply:
https://github.com/ACINQ/eclair/commit/238f51f671daed0ce7d353d341efd063a06cb0d5 looks good to me, we can now remove that added `Try`!

Reply:
🤦‍♂️ 

Done in 2cb3262db.